src/routes/demographics/formatters.js

```diff
@@ -0,0 +1,243 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ 
+ 
+ 
+ /** This Svelte action will format an input element on input, focus, and blur events. 
```

Comment:
Nit, but why does it only format on these events, and not as the user types? Wondering if that is a stylistic choice or just pragmatic. I can see the argument for not modifying input while the user is interacting.

Reply:
I believe `input` is when it formats (which is when the user types / copy-pastes). @changecourse's requirements were to prevent inputs that shouldn't exist (like letters in a number field) instead of showing an error on incorrect input. I tried breaking the experience implemented here in various ways but was unsuccessful. This ultimately felt like the best way to control the nature of the inputs appropriate for (1) the form version of the answers (what should the entered-in format look like when the input field is not active, e.g. `$100,000` is what we see vs `100000` or 100000), (2) the edit version of the answers (we don't want the currency fields while typing to have the dollar sign and commas), and (3) the response version of the answers (what should the data look like when it goes to the servers?) These three things may all be different, hence this formatter.